internal/backend/api.go

```diff
@@ -46,6 +52,27 @@ type ListAccessRequestInput struct {
 	ArgoCDHeaders
 }
 
+// ListAllowedRolesInput defines the input parameters list of allowed roles.
+type ListAllowedRolesInput struct {
+	ArgoCDHeaders
+}
+
+// ListAllowedRolesResponse defines the response of allowed roles requests.
+type ListAllowedRolesResponse struct {
+	Body ListAllowedRolesResponseBody
+}
+
+// ListAllowedRolesResponseBody defines the response body of allowed roles requests.
+type ListAllowedRolesResponseBody struct {
+	Items []AllowedRoleResponseBody `json:"items"`
+}
+
+// AllowedRoleResponseBody defines the allowed role response.
+type AllowedRoleResponseBody struct {
+	RoleName        string `json:"roleName" example:"custom-role-template" doc:"The role template name to request."`
+	RoleDisplayName string `json:"roleDisplayName" example:"Write (DevOps)" doc:"The human friendly name of the role that can be used to display to users."`
```

Comment:
It is probably a good practice to return the ordinal to the client, so they can sort however they want, instead of having a backend sort

Reply:
While I agree with you that this is a good practice, the reasons I implemented this way are:
- Follow the same standard applied by the list AccessResquests operation that brings an ordered list from the backend
- This is a dedicated API built specifically for the EphemeralAccess UI extension.

Anyhow, I can add the order as part of the response if you still think that this is necessary, but I want to keep the current ordering logic in the backend.

Reply:
While I agree with you that this is a good practice, the reasons I implemented this way are:
- Follow the same standard applied by the list AccessResquests operation that brings an ordered list from the backend
- This is a dedicated API built specifically for the EphemeralAccess UI extension.

Anyhow, I can add the order as part of the response if you still think that this is necessary, but I want to keep the current ordering logic in the backend.